src/index.ts

```diff
@@ -17,9 +17,19 @@ export async function run(): Promise<void> {
         // Node 20 introduced automatic family selection for dual-stack endpoints. When the runner 
         // sits far away from the secrets manager endpoint it sometimes timeouts on negotiation between
         // A and AAAA records. This behaviour was described in the https://github.com/nodejs/node/issues/54359
-        // The default value is 250ms, increasing to 1s. The integration tests stops beeing flaky with this
-        // value.
-        setDefaultAutoSelectFamilyAttemptTimeout(1000);
+        // The default value is 1s. We allow configuring this timeout through the
+        // 'auto-select-family-attempt-timeout' parameter to help prevent flaky integration tests
+        
+        const timeout = Number(core.getInput('auto-select-family-attempt-timeout'));
+        
+        if (timeout < 10 || Number.isNaN(timeout)) {
+            core.setFailed(`Invalid value for 'auto-select-family-attempt-timeout': ${timeout}. Must be a number greater than 10.`);
```

Comment:
Nitpick, but technically the condition allows any number that is "greater than or equal to 10", not just "greater than 10"